### PR TITLE
StartProcessingAtom() no longer creates things inside mobs

### LINKED ISSUE #54422: Slicing sliceable food in one hand with a knife in the other deletes the results
See title.

## Reproduction:
- Hold sliceable food
- Hold knife
- Slice food with knife
- Original food and slices disappear

### PR DESCRIPTION
**Pull request by cacogen**
## About The Pull Request

Fixes #54422. 

## Why It's Good For The Game

Slicing a held pizza no longer results in the slices being created inside you.

## Changelog
:cl: cacogen
fix: Slicing food in your hand (e.g. pizza) no longer results in it "disappearing" (i.e. having the slices created inside you)
/:cl:

### CODE REVIEW

**Comment by optimumtact** [comment]
the title of your PR is not correct, since it is in fact still created inside the mob and then moved into their hands

File: code/game/atoms.dm [hunk 1]
```diff
@@ -1258,14 +1258,18 @@
...
 
 /atom/proc/StartProcessingAtom(mob/living/user, obj/item/I, list/chosen_option)
-	to_chat(user, "<span class='notice'>You start working on [src]</span>")
+	to_chat(user, "<span class='notice'>You start working on [src].</span>")
 	if(I.use_tool(src, user, chosen_option[TOOL_PROCESSING_TIME], volume=50))
 		var/atom/atom_to_create = chosen_option[TOOL_PROCESSING_RESULT]
+		var/output_loc = loc
+		if(ismob(loc))
+			var/mob/M = loc
+			output_loc = M.loc
```
Comment (Rohesie) [inline code/game/atoms.dm:1267]:
```suggestion
```

File: code/game/atoms.dm [hunk 2]
```diff
@@ -1258,14 +1258,18 @@
...
 		for(var/i = 1 to chosen_option[TOOL_PROCESSING_AMOUNT])
-			var/atom/created_atom = new atom_to_create(loc)
+			var/atom/created_atom = new atom_to_create(output_loc)
```
Comment (Rohesie) [inline code/game/atoms.dm:1269]:
```suggestion
			var/atom/created_atom = new atom_to_create(get_turf(src))
```

**Comment by cacogen** [comment]
That's a much simpler and less braindead method.

File: code/game/atoms.dm [hunk 3]
```diff
@@ -1268,14 +1268,14 @@
 
 
 /atom/proc/StartProcessingAtom(mob/living/user, obj/item/I, list/chosen_option)
-	to_chat(user, "<span class='notice'>You start working on [src]</span>")
+	to_chat(user, "<span class='notice'>You start working on [src].</span>")
 	if(I.use_tool(src, user, chosen_option[TOOL_PROCESSING_TIME], volume=50))
 		var/atom/atom_to_create = chosen_option[TOOL_PROCESSING_RESULT]
 		for(var/i = 1 to chosen_option[TOOL_PROCESSING_AMOUNT])
-			var/atom/created_atom = new atom_to_create(loc)
+			var/atom/created_atom = new atom_to_create(get_turf(src))
```
Comment (Dennok) [inline code/game/atoms.dm:1275]: Omg. Yse `drop_location()` for this

**Pull request merged**